hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/ComplexKeyGenerator.java

```diff
@@ -18,38 +17,38 @@
 
 package org.apache.hudi.keygen;
 
-import org.apache.hudi.DataSourceWriteOptions;
-import org.apache.hudi.common.config.TypedProperties;
-
 import org.apache.avro.generic.GenericRecord;
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.keygen.constant.KeyGeneratorOptions;
+import org.apache.spark.sql.Row;
 
 import java.util.Arrays;
 import java.util.stream.Collectors;
-import org.apache.spark.sql.Row;
 
 /**
  * Complex key generator, which takes names of fields to be used for recordKey and partitionPath as configs.
  */
 public class ComplexKeyGenerator extends BuiltinKeyGenerator {
 
-  public static final String DEFAULT_RECORD_KEY_SEPARATOR = ":";
+  private CommonComplexKeyGenerator commonComplexKeyGenerator;
```

Comment:
can all these members be final (intellij tip)

Reply:
> can all these members be final (intellij tip)

done